Regression: RegistryTools no longer creates empty `Deps.toml` or `Compat.toml` files

I just tried to register a super simple package of mine, apparently too simple.

It seams to fail because it has no dependencies and the RegistryCI cannot handle this. Here the stacktrace: 
```julia
ERROR: LoadError: "/home/runner/work/General/General/N/NotMacro/Deps.toml": No such file
Stacktrace:
  [1] error(::String, ::String)
    @ Base ./error.jl:44
  [2] readstring
    @ /opt/hostedtoolcache/julia/1.8.5/x64/share/julia/stdlib/v1.8/TOML/src/TOML.jl:21 [inlined]
  [3] parsefile(f::String)
    @ TOML /opt/hostedtoolcache/julia/1.8.5/x64/share/julia/stdlib/v1.8/TOML/src/TOML.jl:43
  [4] meets_compat_for_all_deps(working_directory::String, pkg::String, version::VersionNumber)
    @ RegistryCI.AutoMerge ~/.julia/packages/RegistryCI/51I29/src/AutoMerge/guidelines.jl:83
  [5] (::RegistryCI.AutoMerge.var"#83#84")(data::RegistryCI.AutoMerge.GitHubAutoMergeData)
    @ RegistryCI.AutoMerge ~/.julia/packages/RegistryCI/51I29/src/AutoMerge/guidelines.jl:76
  [6] check!(guideline::RegistryCI.AutoMerge.Guideline, data::RegistryCI.AutoMerge.GitHubAutoMergeData)
    @ RegistryCI.AutoMerge ~/.julia/packages/RegistryCI/51I29/src/AutoMerge/types.jl:138
  [7] pull_request_build(data::RegistryCI.AutoMerge.GitHubAutoMergeData; check_license::Bool)
    @ RegistryCI.AutoMerge ~/.julia/packages/RegistryCI/51I29/src/AutoMerge/pull_requests.jl:211
  [8] pull_request_build(api::GitHub.GitHubWebAPI, pr_number::Int64, current_pr_head_commit_sha::String, registry::GitHub.Repo, registry_head::String; whoami::String, auth::GitHub.OAuth2, authorized_authors::Vector{String}, authorized_authors_special_jll_exceptions::Vector{String}, error_exit_if_automerge_not_applicable::Bool, master_branch::String, master_branch_is_default_branch::Bool, suggest_onepointzero::Bool, registry_deps::Vector{String}, check_license::Bool, public_registries::Vector{String}, read_only::Bool, environment_variables_to_pass::Vector{String})
    @ RegistryCI.AutoMerge ~/.julia/packages/RegistryCI/51I29/src/AutoMerge/pull_requests.jl:166
  [9] run(; env::Base.EnvDict, cicfg::RegistryCI.AutoMerge.GitHubActions, merge_new_packages::Bool, merge_new_versions::Bool, new_package_waiting_period::Day, new_jll_package_waiting_period::Minute, new_version_waiting_period::Minute, new_jll_version_waiting_period::Minute, registry::String, tagbot_enabled::Bool, authorized_authors::Vector{String}, authorized_authors_special_jll_exceptions::Vector{String}, additional_statuses::Vector{String}, additional_check_runs::Vector{String}, error_exit_if_automerge_not_applicable::Bool, master_branch::String, master_branch_is_default_branch::Bool, suggest_onepointzero::Bool, registry_deps::Vector{String}, api_url::String, check_license::Bool, public_registries::Vector{String}, read_only::Bool, environment_variables_to_pass::Vector{String})
    @ RegistryCI.AutoMerge ~/.julia/packages/RegistryCI/51I29/src/AutoMerge/public.jl:141
 [10] top-level scope
    @ ~/work/_temp/2e56103b-ce75-4e54-b2cd-9c7e89bc80[72](https://github.com/JuliaRegistries/General/actions/runs/3927097124/jobs/6713411399#step:16:73):4
in expression starting at /home/runner/work/_temp/2e56103b-ce[75](https://github.com/JuliaRegistries/General/actions/runs/3927097124/jobs/6713411399#step:16:76)-4e54-b2cd-9c7e89bc[80](https://github.com/JuliaRegistries/General/actions/runs/3927097124/jobs/6713411399#step:16:81)72:4
Error: Process completed with exit code 1.
```

I would be very thankful if someone could manually merge [the registration pullrequest which failed](https://github.com/JuliaRegistries/General/pull/75744) in the meantime.

--- a/src/register.jl
+++ b/src/register.jl
@@ -387,7 +387,14 @@ function update_deps_file(pkg::Project,
         end
 
         deps_data[pkg.version] = deps
-        if !isempty(deps_data) && !all(isempty, values(deps_data))
+        if file == "Deps.toml"
+            # We always write the `Deps.toml` file, even if it is empty
+            write_this_file = true
+        else
+            # We only write the `WeakDeps.toml` file if it is not empty
+            write_this_file = !isempty(deps_data) && !all(isempty, values(deps_data))
+        end
+        if write_this_file
             Compress.save(deps_file, deps_data)
         end
     end
@@ -491,7 +498,14 @@ function update_compat_file(pkg::Project,
         end
 
         compat_data[pkg.version] = d
-        if !isempty(compat_data) && !all(isempty, values(compat_data))
+        if file == "Compat.toml"
+            # We always write the `Compat.toml` file, even if it is empty
+            write_this_file = true
+        else
+            # We only write the `WeakCompat.toml` file if it is not empty
+            write_this_file = !isempty(compat_data) && !all(isempty, values(compat_data))
+        end
+        if write_this_file
             Compress.save(compat_file, compat_data)
         end
     end

